server/methods/eraseRoom.js

```diff
@@ -17,7 +17,7 @@ Meteor.methods({
 			});
 		}
 
-		if (RocketChat.authz.hasPermission(fromId, `delete-${ room.t }`, rid)) {
+		if (RocketChat.authz.hasPermission(fromId, `delete-${ room.t }`, rid)||automatic===true) {
```

Comment:
The correct way to do that is separate the content of this method without the permission check to a new function and call that function here and inside your method. We did this in some methods like this one https://github.com/RocketChat/Rocket.Chat/blob/HEAD/server/methods/deleteUser.js#L35